### PR TITLE
Refactor the RollCallHandler functions

### PR DESCRIPTION
**Pull request by Ajkunas**
I have tried to refactor as we discussed during the meeting: 

- I created a function `checkParameters` in `MessageHandler` file, so that we avoid having several matching cases (that is the simpler way I found). Also, this function can be used in every handler file. 
- Concerning the ChannelExists function, I did not find a simple way to avoid using the `transformWith` function. Any suggestion could be useful 
- In line 80 of `RollCallHandler` file, I avoided the `transformWith` function, because I thought it was not really useful, since we already get a error message from the function `ChannelExists` itself. 
- Concerning the `CloseRollCall` function, I rearranged the function by redefining the function `createAttendeeChannels`  and putting it outside as a private function.

### CODE REVIEW

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala
Comment (tuomas-p) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala:22]: looks way cleaner, as we supposed!

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (tuomas-p) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:137]: good point in isolating this

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (tuomas-p) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:110]: I don't know how much you've already thought about it, the function looks already way cleaner but maybe it could be stripped down a bit more still, I'll try to figure out something

**Review by tuomas-p — COMMENTED** [review]
Clean refactoring! I'll also try to look up something to go further but that's already way more understandable imo

**Review by pierluca — COMMENTED** [review]
I had a quick look, nothing to add: it's indeed much better 👍 , and you've already highlighted the few spots which can still be improved 👍

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (Ajkunas) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala]: I added a new function `checkLaoChannel` which does the same as `checkParameters`.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (Ajkunas) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:46]: I agree that both lines repeat, but I tried to figure out how to do that, and could not find a proper way without adding a line which checks if the `checkParameters` function throws an error, and therefore we cannot access to the message itself. So, the way as I did it seemed cleaner for me.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:54]: would it make sense to call this `AssertChannelMissing` ?
It might more clearly convey the idea that this WILL fail if our expectation is not met.

**Review by pierluca — COMMENTED** [review]
To my Scala-newbie eyes, this looks really great. Good job! 😁 👍

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala
Comment (tuomas-p) [inline be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala:110]:
```suggestion
  test("CreateRollCall should fail if the rollcall already exists in database") {
```

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala
Comment (tuomas-p) [inline be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala:102]:
```suggestion
  test("CreateRollCall should succeed if the rollcall doesn't already exist in the database") {
```

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala
Comment (tuomas-p) [inline be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandlerTest.scala:158]: nice additions

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (tuomas-p) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:49]: Would you mean that this newly-created polymorphic method would output both the message and the decodedData at once, failing if one of the steps fails?

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala
Comment (tuomas-p) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:54]: Good point in my opinion

Comment (Ajkunas) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala:54]: Yes, I agree with this point as well

**Review by pierluca — COMMENTED** [review]
I'll let you guys review it and approve it if appropriate, but I have to say it looks so much cleaner 😍 
Very good job @Ajkunas !

**Review by tuomas-p — APPROVED** [review]
For me the PR is now good to go, I'll create a PR based off your branch for the polymorphic function as discussed on Slack unless you want to merge it directly

**Pull request merged**
